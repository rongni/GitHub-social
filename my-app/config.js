export const GITHUB_ACCESS_TOKEN = '6b5e835e5b74e13df0e91a6393101fd1d20c81ae';
