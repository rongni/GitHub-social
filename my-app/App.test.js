jest.useFakeTimers();
import React from 'react';
import renderer from 'react-test-renderer';
import Home from './Components/home';
import User from './Components/user_profile';
import Repo from './Components/repo';
import Following from './Components/following';
import Follower from './Components/follower';
const navigation = { navigate: jest.fn() };
import fetch from 'node-fetch';
import { ApolloProvider } from 'react-apollo';
import { ApolloClient } from 'apollo-boost';
import { ApolloLink } from 'apollo-link';
import { createHttpLink } from 'apollo-link-http';
import { InMemoryCache } from 'apollo-cache-inmemory';
import Router from './Components/router';
import { GITHUB_ACCESS_TOKEN } from './config';
const navigation_two = {
	state: {
		params: { username: 'rongni' },
	},
};
function createClient() {
	const httpLink = createHttpLink({
		uri: 'https://api.github.com/graphql',
		fetch,
	});

	// apply widdleware to add access token to request
	let middlewareLink = new ApolloLink((operation, forward) => {
		operation.setContext({
			headers: {
				authorization: `Bearer ${GITHUB_ACCESS_TOKEN}`,
			},
		});
		return forward(operation);
	});
	const link = middlewareLink.concat(httpLink);

	// Initialize Apollo Client with URL to our server
	return new ApolloClient({
		link: link,
		cache: new InMemoryCache(),
	});
}

it('renders correctly', () => {
	const component = renderer.create(
		<ApolloProvider client={createClient()}>
			<Repo navigation={navigation_two} />
		</ApolloProvider>
	);

	const tree = component.toJSON();
	expect(tree).toMatchSnapshot();
});
it('renders correctly', () => {
	const component = renderer.create(
		<ApolloProvider client={createClient()}>
			<User searchQuery='rongni' navigation={navigation} />
		</ApolloProvider>
	);

	const tree = component.toJSON();
	expect(tree).toMatchSnapshot();
});

it('renders correctly', () => {
	const rendered = renderer.create(<Home navigation={navigation} />).toJSON();
	expect(rendered).toMatchSnapshot();
});
it('renders correctly', () => {
	const rendered = renderer.create(<Follower />).toJSON();
	expect(rendered).toMatchSnapshot();
});
it('renders correctly', () => {
	const rendered = renderer.create(<Following />).toJSON();
	expect(rendered).toMatchSnapshot();
});
describe('<Follower />', () => {
	it('has 1 child', () => {
		const tree = renderer.create(<Follower />).toJSON();
		expect(tree.children.length).toBe(1);
	});
});

describe('<Following />', () => {
	it('has 1 child', () => {
		const tree = renderer.create(<Following />).toJSON();
		expect(tree.children.length).toBe(1);
	});
});

describe('<Home />', () => {
	it('has 2 child', () => {
		const tree = renderer.create(<Home navigation={navigation} />).toJSON();
		expect(tree.children.length).toBe(2);
	});
});

describe('<User />', () => {
	it('has 1 child', () => {
		const tree = renderer
			.create(
				<ApolloProvider client={createClient()}>
					<User searchQuery='rongni' navigation={navigation} />
				</ApolloProvider>
			)
			.toJSON();
		expect(tree.children.length).toBe(1);
	});
});
describe('<Repo />', () => {
	it('has 1 child', () => {
		const tree = renderer
			.create(
				<ApolloProvider client={createClient()}>
					<Repo navigation={navigation_two} />
				</ApolloProvider>
			)
			.toJSON();
		expect(tree.children.length).toBe(1);
	});
});

describe('<Profile />', () => {
	it('loading', () => {
		const tree = renderer
			.create(
				<ApolloProvider client={createClient()}>
					<User searchQuery='rongni' navigation={navigation} />
				</ApolloProvider>
			)
			.toJSON();
		expect(tree.children).toStrictEqual(['fetching posts... ']);
	});
});

describe('<Repository />', () => {
	it('loading ', () => {
		const tree = renderer
			.create(
				<ApolloProvider client={createClient()}>
					<Repo navigation={navigation_two} />
				</ApolloProvider>
			)
			.toJSON();
		expect(tree.children).toStrictEqual(['fetching posts... ']);
	});
});
