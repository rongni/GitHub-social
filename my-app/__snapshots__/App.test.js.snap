// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<Text>
  fetching posts... 
</Text>
`;

exports[`renders correctly 2`] = `
<Text>
  fetching posts... 
</Text>
`;

exports[`renders correctly 3`] = `
<View
  collapsable={false}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "rgb(231, 231, 231)",
          "borderTopLeftRadius": 2,
          "borderTopRightRadius": 2,
        },
        Object {
          "justifyContent": "center",
        },
      ]
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "rgba(0, 0, 0, 0.26)",
          "bottom": 0,
          "height": 2,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "transform": Array [
            Object {
              "scaleY": 0.5,
            },
          ],
        }
      }
    />
    <View
      style={
        Array [
          Object {
            "paddingBottom": 0,
            "paddingTop": 0,
          },
          Object {
            "minHeight": 60,
          },
        ]
      }
    >
      <View
        pointerEvents="none"
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "opacity": 1,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "transform": Array [
              Object {
                "translateX": 0,
              },
            ],
          }
        }
      >
        <Text
          numberOfLines={1}
          onLayout={[Function]}
          style={
            Object {
              "color": "#3498db",
              "fontFamily": "System",
              "fontSize": 16,
              "fontWeight": undefined,
              "left": 0,
              "opacity": 0,
              "paddingLeft": 12,
              "paddingRight": 12,
              "position": "absolute",
              "textAlign": "left",
              "top": 30,
              "transform": Array [
                Object {
                  "translateX": 0,
                },
                Object {
                  "translateY": 0,
                },
                Object {
                  "scale": 1,
                },
              ],
              "writingDirection": "ltr",
            }
          }
        >
          Search Github
        </Text>
        <Text
          numberOfLines={1}
          style={
            Object {
              "color": "rgba(0, 0, 0, 0.54)",
              "fontFamily": "System",
              "fontSize": 16,
              "fontWeight": undefined,
              "left": 0,
              "opacity": 0,
              "paddingLeft": 12,
              "paddingRight": 12,
              "position": "absolute",
              "textAlign": "left",
              "top": 30,
              "transform": Array [
                Object {
                  "translateX": 0,
                },
                Object {
                  "translateY": 0,
                },
                Object {
                  "scale": 1,
                },
              ],
              "writingDirection": "ltr",
            }
          }
        >
          Search Github
        </Text>
      </View>
      <TextInput
        allowFontScaling={true}
        editable={true}
        multiline={false}
        onBlur={[Function]}
        onChangeText={[Function]}
        onFocus={[Function]}
        placeholder=""
        placeholderTextColor="rgba(0, 0, 0, 0.54)"
        rejectResponderTermination={true}
        selectionColor="#3498db"
        style={
          Array [
            Object {
              "flexGrow": 1,
              "margin": 0,
              "zIndex": 1,
            },
            Object {
              "paddingLeft": 12,
              "paddingRight": 12,
            },
            Object {
              "height": 60,
            },
            Object {
              "paddingBottom": 4,
              "paddingTop": 24,
            },
            Object {
              "color": "#000000",
              "fontFamily": "System",
              "fontSize": 16,
              "fontWeight": undefined,
              "textAlign": "left",
              "textAlignVertical": "center",
            },
            Array [
              Object {},
            ],
          ]
        }
        textAlign="center"
        underlineColorAndroid="transparent"
        value=""
      />
    </View>
  </View>
  <View
    style={
      Object {
        "backgroundColor": "transparent",
        "borderColor": "transparent",
        "borderRadius": 2,
        "borderStyle": "solid",
        "borderWidth": 0,
        "elevation": 0,
        "minWidth": 64,
      }
    }
  >
    <View
      accessibilityRole="button"
      accessibilityState={
        Object {
          "disabled": undefined,
        }
      }
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "overflow": "hidden",
          },
          Object {
            "borderRadius": 2,
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "center",
            },
            undefined,
          ]
        }
      >
        <Text
          numberOfLines={1}
          selectable={false}
          style={
            Array [
              Object {
                "color": "#000000",
                "fontFamily": "System",
                "fontWeight": "400",
              },
              Object {
                "textAlign": "left",
              },
              Array [
                Object {
                  "letterSpacing": 1,
                  "marginHorizontal": 16,
                  "marginVertical": 9,
                  "textAlign": "center",
                },
                undefined,
                Object {
                  "textTransform": "uppercase",
                },
                Object {
                  "color": "#3498db",
                  "fontFamily": "System",
                  "fontWeight": "500",
                },
                Object {
                  "fontFamily": "System",
                  "fontWeight": "500",
                },
                undefined,
              ],
            ]
          }
        >
          Search
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`renders correctly 4`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View>
    <Text>
      Follower is here
    </Text>
  </View>
</View>
`;

exports[`renders correctly 5`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View>
    <Text>
      Following is here
    </Text>
  </View>
</View>
`;
